src/main/resources/static/api.yaml

```diff
@@ -88,4 +172,122 @@ components:
\ No newline at end of file
+          description: Identifier of the health information provider where patient record needs to be looked up.
+
+    ConsentRequestsRepresentation:
+      type: object
+      properties:
+        limit:
+          type: integer
+        size:
+          type: integer
+        offset:
+          type: integer
+        requests:
+          type: array
+          maxItems: 100
+          items:
+            $ref: '#/components/schemas/ConsentRequestRepresentation'
+
+    ConsentRequestRepresentation:
+      type: object
+      properties:
+        purpose:
+          type: object
+          properties:
+            description:
+              type: string
+        hitypes:
+          type: array
+          items:
+            $ref: '#/components/schemas/HITypeRepresentation'
+        requester:
+          type: object
+          properties:
+            id:
+              type: string
+              format: uuid
+              example: 496413f0-7b63-4f80-9bb0-9382cc3108d3
+            name:
+              type: string
+              example: "Dr. Karuppan"
+            orgName:
+              type: string
+              example: "TMH"
+            contact:
+              type: object
+              properties:
+                mobile:
+                  type: string
+                  example: '+9199999x9999'
+        duration:
+          type: object
+          properties:
+            from:
+              type: string
+              format: date-time
+            to:
+              type: string
+              format: date-time
+        expiry:
+          type: string
+          format: date-time
+        status:
+          type: string
+          enum:
+            - REQUESTED
+            - EXPIRED
+            - DENIED
+            - GRANTED
+
+    HITypeRepresentation:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/HITypeEnum'
+        description:
+          type: string
+          example: HIType Description2
+
+    HITypeEnum:
+      type: string
+      enum:
+        - HIType1
+        - HIType2
+        - HIType3
+        - HIType4
+        - HIType5
+
+    ConsentApproveRequest:
+      type: object
+      properties:
+        consent:
+          type: object
+          properties:
+            hitypes:
+              type: array
+              items:
+                $ref: '#/components/schemas/HITypeEnum'
+            informationProviders:
+              type: array
+              items:
+                type: object
+                properties:
+                  id:
+                    type: string
+                    description: Identifier of the information provider.
+                  carecontexts:
+                    type: array
+                    description: List of care context ids.
+                    items:
+                      type: string
+                      description: care context id.
+            duration:
```

Comment:
duration can go inside "permission", in the form
"permission": {
      "accessMode": "VIEW",
      "dateRange": {
        "from": "2020-01-03T13:25:15.550Z",
        "to": "2020-01-03T13:25:15.551Z"
      },
      "dataExpiryAt": "2020-01-03T13:25:15.551Z",
      "frequency": {
        "unit": "HOUR",
        "value": 2
      }
  },